lib/elixir/lib/io.ex

```diff
@@ -36,7 +36,7 @@ defmodule IO do
 
   @type device :: atom | pid
   @type nodata :: {:error, term} | :eof
-  @type chardata() :: :unicode.chardata()
+  @type chardata :: charlist | binary
```

Comment:
Those are not equivalent. A chardata can contain a list of binaries, for example. :)

Reply:
@josevalim Oops - I got a little confused reading the Erlang docs because they have a `unicode.charlist` type!